exp/controllers/azuremanagedmachinepool_controller.go

```diff
@@ -188,6 +195,12 @@ func (r *AzureManagedMachinePoolReconciler) reconcileNormal(ctx context.Context,
 	}
 
 	if err := newAzureManagedMachinePoolReconciler(scope).Reconcile(ctx, scope); err != nil {
+		if IsAgentPoolVMSSNotFoundError(err) {
+			// if the underlying VMSS is not yet created, requeue for 30s in the future
+			return reconcile.Result{
+				RequeueAfter: 30 * time.Second,
```

Comment:
why 30s here instead of allowing exponential backoff? is this to partially account for https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/847?

Reply:
The duration is arbitrary, but the idea was to give it some time and try again to allow the AzureManagedControlPlane a chance to create the AKS cluster.